tests/pytests/test_aggregate_count.py

```diff
@@ -0,0 +1,296 @@
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LIMIT', 0, 0], docs),
+        (['FT.AGGREGATE', 'games', '*', 'WITHOUTCOUNT', 'LIMIT', 0, 0], ANY),
+        (['FT.AGGREGATE', 'games', '*', 'LIMIT', 0, 0], ANY),
+    ]
+
+    for query, expected_results in queries_and_results:
+        cmd=' '.join(str(x) for x in query)
+        for dialect in [1, 2, 3, 4]:
+            env.expect('CONFIG', 'SET', 'search-default-dialect', dialect).ok()
+            res = env.cmd(*query)
+            total_results = _get_total_results(res)
+            results = _get_results(res)
+
+            # Verify results
+            env.assertEqual(total_results, expected_results,
+                            message=f'{cmd}: total_results != expected')
+            env.assertEqual(len(results), 0,
+                            message=f'{cmd}: len(results) != 0')
+
+
+def test_limit00_resp3():
+    _test_limit00(3)
+
+
+def test_limit00_resp2():
+    _test_limit00(2)
+
+
+def _test_withcount(protocol):
+    env = Env(protocol=protocol)
+    add_values_iterations = 1
+    _setup_index_and_data(env, number_of_iterations=add_values_iterations)
+    docs = 2265 * add_values_iterations
+
+    queries_and_results = [
+        # WITHCOUNT
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT'], docs),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LIMIT', 0, 50], 50),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LIMIT', 0, int(docs/2)], int(docs/2)),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LIMIT', 0, docs*4], docs),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LIMIT', 10, 50], 50),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LIMIT', 100, docs], docs - 100),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LIMIT', docs, docs*2], 0),
+        # WITHCOUNT + LOAD
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LOAD', '1', '@title'], docs),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'LOAD', '1', '@price'], docs),
+        # WITHCOUNT + SORTBY
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'SORTBY', '1', '@title'], 2265),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'SORTBY', '1', '@price'], 2265),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'SORTBY', '2', '@title', 'ASC'], 2265),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'SORTBY', '2', '@price', 'ASC'], 2265),
+        # WITHCOUNT + SORTBY + LIMIT
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 0, 50], 50),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 0, int(docs/2)], int(docs/2)),
+        (['FT.AGGREGATE', 'games', '*', 'WITHCOUNT', 'SORTBY', 1, '@title', 'LIMIT', 0, docs*4], docs),
+    ]
+
+    for query, expected_results in queries_and_results:
+        cmd=' '.join(str(x) for x in query)
+        for dialect in [1, 2, 3, 4]:
+            env.expect('CONFIG', 'SET', 'search-default-dialect', dialect).ok()
```

Comment:
Notice you'll need to protect `search-default-dialect` with `enable-unstable-features` in the backports..
(Not a comment just for you to notice 😉)